### ISSUE TITLE
To sort an ImList using Comparable

### ISSUE THREAD

**Issue by imkwokyong** [nus-cs2030/2223-s2#153]
In the documentation of ImList:

![image](https://user-images.githubusercontent.com/110521168/219613630-0e313324-9789-4c57-a730-9ef2f9f9fac3.png)

If I want to sort an ImList by the Comparable that I have implemented in the class, I need to do:
`list.sort(null)`

But `null` value is generally not allowed, so how do I sort the ImList by Comparable?

**Comment by Xequil1** [comment]
I created a Booking Comparator that uses the Booking compareTo() function.

**Comment by JustinGohkh** [comment]
You have to create a booking comparator that calls the compareTo() function implemented in booking, and then use the comparator as the params in the sort.

**Comment by imkwokyong** [comment]
Yep, just wondering if it's possible to directly sort ImList using Comparable, similar to an ArrayList.

**Comment by trwstin** [comment]
> Yep, just wondering if it's possible to directly sort ImList using Comparable, similar to an ArrayList.

If you read the javadocs for ImList, it does have a sort method. The sort method can take in a Comparator as a parameter. 

However, if you leave it empty, it will use the Comparable of the Object class of the ImList. Hence the question suggesting the use of implementing Comparable in the Booking class.

**Comment by imkwokyong** [comment]
If I leave it empty, it raises an exception for missing arguments.

**Comment by trwstin** [comment]
> If I leave it empty i.e. `list.sort()`, it raises an exception for missing arguments.

Aw man, hmmm it does say something along the lines of if specified Comparator is {@code null}, then it will use the natural ordering of the object based on its Comparable... Unless this "@code null" does not refer to leaving it empty? 🤔

**Comment by imkwokyong** [comment]
I believe "@code null" means actually doing `list.sort(null)`. I tried in my own computer and it works.
